contrib/verify-commits/README.md

```diff
@@ -24,3 +24,24 @@ keys:
 Note that the above isn't a good UI/UX yet, and needs significant improvements
 to make it more convenient and reduce the chance of errors; pull-reqs
 improving this process would be much appreciated.
+
+Configuration files
+-------------------
+
+* `trusted-git-root`: This file should contain a single git commit hash which is the first unsigned git commit (hence it is the "root of trust").
+* `trusted-sha512-root-commit`: This file should contain a single git commit hash which is the first commit without a SHA512 root commitment.
+* `trusted-keys`: This file should contain a \n-delimited list of all PGP fingerprints of authorized commit signers (primary, not subkeys).
+* `allow-revsig-commits`: This file should contain a \n-delimited list of git commit hashes. See next section for more info.
+
+Key expiry/revocation
+---------------------
+
+When a key (or subkey) which has signed old commits expires or is revoked,
+verify-commits will start failing to verify all commits which were signed by
+said key. In order to avoid bumping the root-of-trust `trusted-git-root`
+file, individual commits which were signed by such a key can be added to the
+`allow-revsig-commits` file. That way, the PGP signatures are still verified
```

Comment:
q: Is there a command to get a list of commits signed by a certain key? If so makes sense to mention that here.

Reply:
Not currently, I generated the diff here with this patch:

```
diff --git a/contrib/verify-commits/verify-commits.sh b/contrib/verify-commits/verify-commits.sh
index 532b97a438..eb41d90941 100755
--- a/contrib/verify-commits/verify-commits.sh
+++ b/contrib/verify-commits/verify-commits.sh
@@ -62,17 +62,22 @@ while true; do
        fi
 
        if ! git -c "gpg.program=${DIR}/gpg.sh" verify-commit "$CURRENT_COMMIT" > /dev/null; then
-               if [ "$PREV_COMMIT" != "" ]; then
-                       echo "No parent of $PREV_COMMIT was signed with a trusted key!" > /dev/stderr
-                       echo "Parents are:" > /dev/stderr
-                       PARENTS=$(git show -s --format=format:%P $PREV_COMMIT)
-                       for PARENT in $PARENTS; do
-                               git show -s $PARENT > /dev/stderr
-                       done
+               export BITCOIN_VERIFY_COMMITS_ALLOW_REVSIG=1
+               if git -c "gpg.program=${DIR}/gpg.sh" verify-commit "$CURRENT_COMMIT" > /dev/null; then
+                       echo "$CURRENT_COMMIT" > /dev/stderr
                else
-                       echo "$CURRENT_COMMIT was not signed with a trusted key!" > /dev/stderr
+                       if [ "$PREV_COMMIT" != "" ]; then
+                               echo "No parent of $PREV_COMMIT was signed with a trusted key!" > /dev/stderr
+                               echo "Parents are:" > /dev/stderr
+                               PARENTS=$(git show -s --format=format:%P $PREV_COMMIT)
+                               for PARENT in $PARENTS; do
+                                       git show -s $PARENT > /dev/stderr
+                               done
+                       else
+                               echo "$CURRENT_COMMIT was not signed with a trusted key!" > /dev/stderr
+                       fi
+                       exit 1
                fi
-               exit 1
        fi
 
        # We always verify the top of the tree
```

Reply:
Ah yes, patching the tool to print them out makes sense.